lib/node/rpc.js

```diff
@@ -714,6 +716,21 @@ class RPC extends RPCBase {
     return await this.headerToJSON(entry);
   }
 
+  async getBlockFilter(args, help) {
+    if (help || args.length < 1 || args.length > 2)
+      throw new RPCError(errs.MISC_ERROR, 'getblockfilter "hash"');
+
+    const valid = new Validator(args);
+    const hash = valid.brhash(0);
+
+    if (!hash)
+      throw new RPCError(errs.MISC_ERROR, 'Invalid block hash.');
+
+    const filter = await this.node.getBlockFilter(hash);
+
```

Comment:
Should check if the block filter is found, and error similar to `getBlock`:
```patch
     const filter = await this.node.getBlockFilter(hash);
 
+    if (!filter)
+      throw new RPCError(errs.MISC_ERROR, 'Block filter not found.');
+
     return filter.toJSON();
   }

```

Reply:
Fixed. Also added error message when indexing is disabled.

Reply:
I know I'm late here, but this RPC doesn't exactly match the bitcoind RPC:

https://github.com/bitcoin/bitcoin/blob/796b71363396e2ac99d241f5975c0978cdae3d67/src/rpc/blockchain.cpp#L2175

...which accepts a second argument for filter type and at least checks it for validity. 

In the event that a new type is added in bitcoind before implemented in bcoin, a bcoin user might request that type and bcoin would silently ignore it and return the `'basic'` filter by default.

Reply:
Okay, good to note.